source/extensions/transport_sockets/alts/config.cc

```diff
@@ -26,6 +26,15 @@ using GrpcAltsCredentialsOptionsPtr =
 
 namespace {
 
+// There is an equivalent function grpc_alts_set_rpc_protocol_versions in newer
+// release of gRPC that should be called directly when available.
+void grpcAltsSetRpcProtocolVersions(grpc_gcp_rpc_protocol_versions* rpc_versions) {
```

Comment:
@tgalkovskyi is there any way to verify this in the integration test I sent you offline? If it's possible to do without feats of engineering then I think we should, so that we don't regress. Also, if you can link to some docs which explain that we should be setting this it would be great.
/wait

Reply:
@htuch documentation wise this design doc is the best I've found so far with the help of internal team who designed ALTS: https://docs.google.com/document/d/1kP6o0tqiSshhWpad7YLzTCHj-4LYcr1zRK4QGdyDP28/edit#heading=h.gijrsdjw51q5

from what I've learned rpc version wasn't set from the beginning because the use case they had in mind was envoy-to-envoy communication where both parties would thus simply ignore the rpc versions and do not perform a check. However this assumption break when you consider envoy-to-grpc client/server communication as this check can be performed by the other party.

I will separately post my musings about test - I think it should be feasible.

Reply:
Ack. FWIW that's a Google-internal only doc. It would be ideal to have ALTS compliance fully specced somewhere :-D 

What you're doing here seems totally legit; LMK where you end up with in terms of testablity. I don't think we expect gymnastics, but if it can be done reasonably let's do it.